blueocean-dashboard/src/main/java/io/jenkins/blueocean/preload/FavoritesStatePreloader.java

```diff
@@ -0,0 +1,83 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.blueocean.preload;
+
+import hudson.Extension;
+import hudson.model.User;
+import io.jenkins.blueocean.commons.BlueUrlTokenizer;
+import io.jenkins.blueocean.commons.RESTFetchPreloader;
+import io.jenkins.blueocean.commons.stapler.ModelObjectSerializer;
+import io.jenkins.blueocean.rest.Reachable;
+import io.jenkins.blueocean.rest.model.BlueFavorite;
+import io.jenkins.blueocean.rest.model.BlueFavoriteContainer;
+import io.jenkins.blueocean.service.embedded.rest.UserImpl;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Preload user favorites.
+ *
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+@Extension
+public class FavoritesStatePreloader extends RESTFetchPreloader {
+
+    private static final Logger LOGGER = Logger.getLogger(FavoritesStatePreloader.class.getName());
+
+    @Override
+    protected FetchData getFetchData(@Nonnull BlueUrlTokenizer blueUrl) {
+        User jenkinsUser = User.current();
+
+        if (jenkinsUser != null) {
+            UserImpl blueUser = new UserImpl(jenkinsUser);
+            BlueFavoriteContainer favoritesContainer = blueUser.getFavorites();
+
+            if (favoritesContainer != null) {
+                JSONArray favorites = new JSONArray();
+                Iterator<BlueFavorite> favoritesIterator = favoritesContainer.iterator();
+
+                while(favoritesIterator.hasNext()) {
+                    Reachable favorite = favoritesIterator.next();
+                    try {
+                        favorites.add(JSONObject.fromObject(ModelObjectSerializer.toJson(favorite)));
+                    } catch (IOException e) {
+                        LOGGER.log(Level.FINE, String.format("Unable to preload favorites for User '%s'. Serialization error.", jenkinsUser.getFullName()), e);
+                        return null;
+                    }
+                }
+
+                return new FetchData(favoritesContainer.getLink().getHref(), favorites.toString());
+            }
+        }
+
+        // Don't preload any data on the page.
+        return null;
+    }
```

Comment:
Yeah, this is one of those subjective/personal taste things. I prefer it as it is because, if split up, you'd then find that you'd have some code duplication between the `shouldPreload` and the `getFetchData` because the `getFetchData` would need to get many of the same things as the `shouldPreload` in order to get at the data to be preloaded (e.g. looking up runs on jobs etc). Of course you'd then say that you could split out helper methods etc, but that's just adding more conceptual weight to the whole thing, not to mention the fact that it would result in the same code searches happening multiple times per request, unless you start making these preloaders stateful etc, and that's def something we do not need to be doing.

So @cliffmeyers, if it's all the same with you, I'd prefer to leave that as is.